### PR TITLE
Add include and exclude filter to environment registry

### PR DESCRIPTION
**Pull request by cazfletch**
Added filters to environment registry to tidy up getting all environments

Signed-off-by: Caroline Fletcher <cazfletch@gmail.com>

### CODE REVIEW

File: packages/blockchain-extension/extension/commands/addGatewayCommand.ts [hunk 1]
```diff
@@ -48,12 +49,12 @@ export async function addGateway(): Promise<{} | void> {
 
         if (gatewayMethod === UserInputUtil.ADD_GATEWAY_FROM_ENVIRONMENT) {
 
-            const environments: FabricEnvironmentRegistryEntry[] = await FabricEnvironmentRegistry.instance().getAll(false, false, true);
+            const environments: FabricEnvironmentRegistryEntry[] = await FabricEnvironmentRegistry.instance().getAll([], [EnvironmentFlags.ANSIBLE]);
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/extension/commands/addGatewayCommand.ts]: Does this exclude all Ansible environments?

File: packages/blockchain-extension/extension/commands/deleteEnvironmentCommand.ts [hunk 2]
```diff
@@ -36,13 +36,13 @@ export async function deleteEnvironment(environment: FabricEnvironmentTreeItem |
...
             // Get all environments, including local environments.
-            const environments: Array<FabricEnvironmentRegistryEntry> = await FabricEnvironmentRegistry.instance().getAll(true);
+            const environments: Array<FabricEnvironmentRegistryEntry> = await FabricEnvironmentRegistry.instance().getAll();
             if (environments.length === 0) {
                 outputAdapter.log(LogType.ERROR, `No environments to delete.`);
                 return;
             }
 
-            const chosenEnvironment: IBlockchainQuickPickItem<FabricEnvironmentRegistryEntry>[] = await UserInputUtil.showFabricEnvironmentQuickPickBox('Choose the environment(s) that you want to delete', true, false, true) as IBlockchainQuickPickItem<FabricEnvironmentRegistryEntry>[];
+            const chosenEnvironment: IBlockchainQuickPickItem<FabricEnvironmentRegistryEntry>[] = await UserInputUtil.showFabricEnvironmentQuickPickBox('Choose the environment(s) that you want to delete', true, false, []) as IBlockchainQuickPickItem<FabricEnvironmentRegistryEntry>[];
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/extension/commands/deleteEnvironmentCommand.ts]: I don't think you don't need to have `[]` as it's the default value

File: packages/blockchain-extension/extension/commands/addGatewayCommand.ts (hunk 1 above)
Comment (cazfletch) [reply packages/blockchain-extension/extension/commands/addGatewayCommand.ts]: yes

File: packages/blockchain-extension/test/commands/deleteEnvironmentCommand.test.ts [hunk 3]
```diff
@@ -123,8 +125,8 @@ describe('DeleteEnvironmentCommand', () => {
 
             await vscode.commands.executeCommand(ExtensionCommands.DELETE_ENVIRONMENT);
 
-            showFabricEnvironmentQuickPickBoxStub.should.have.been.calledOnceWithExactly('Choose the environment(s) that you want to delete', true, false, true);
-            getAllSpy.should.have.been.calledOnceWithExactly(true);
+            showFabricEnvironmentQuickPickBoxStub.should.have.been.calledOnceWithExactly('Choose the environment(s) that you want to delete', true, false, []);
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/test/commands/deleteEnvironmentCommand.test.ts]: I don't think you need `[]` in any of these either - as you're saying "get me all environments" (and the default values are []).

File: packages/blockchain-extension/extension/commands/addGatewayCommand.ts [hunk 4]
```diff
@@ -48,12 +49,12 @@ export async function addGateway(): Promise<{} | void> {
...
             if (environments.length === 0) {
                 throw new Error(`No environments to choose from. Gateways cannot be created from managed Ansible or local environments.`);
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/extension/commands/addGatewayCommand.ts]: `No environments to choose from. Gateways cannot be created from Ansible or local environments.`

**Pull request merged**